tests/warn_conflict/tests/01-make.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import subprocess
+
+BOARD = os.getenv('BOARD', 'stm32f4discovery')
+
+
+def testfunc():
+    cross_gcc = "arm-none-eabi-gcc"
+
+    try:
+        devnull = open(os.devnull)
+        subprocess.Popen([cross_gcc],
+                         stdout=devnull, stderr=devnull).communicate()
+        output = subprocess.Popen(["make"],
+                                  env=os.environ,
```

Comment:
What I suggested was

```
env = os.environ
env["BOARD"] = "stm32f4discovery"
output = subprocess.Popen(["make"], env=env, stderr=subprocess.PIPE, stdout=subprocess.PIPE )
```

But since you are doing a complete rewrite anyways: why not use `pexpect` instead (won't work with `testrunner` though)?

Reply:
I tested your suggestion, e.g only passing the BOARD env to the subprocess but it's not working because of the colored output. I could only make it work by passing the whole env.

Regarding the testrunner, I don't think it possible since we are not use the serial output but only the build command output. It might be possible but it would also require to modify the testrunner.

Reply:
> Regarding the testrunner, I don't think it possible since we are not use the serial output but only the build command output. It might be possible but it would also require to modify the testrunner.

Please read what I write! I said you won't be able to use `testrunner`, but you can still use "raw" `pexpect`.

Reply:
> Please read what I write!

Oups sorry ... Yes that makes sense to use pexpect, I'll try that later.

Reply:
@miri64 my last commit is now using pexpect.